src/main/kotlin/nl/info/zac/app/zaak/ZaakKoppelenRestService.kt

```diff
@@ -115,7 +116,12 @@ class ZaakKoppelenRestService @Inject constructor(
         (areBothOpen(sourceZaak, targetZaak) || areBothClosed(sourceZaak, targetZaak)) &&
             sourceZaak.hasLinkRights() &&
             targetZaak.hasLinkRights() &&
-            targetZaak.isLinkableTo(sourceZaak, relationType)
+            targetZaak.isLinkableTo(sourceZaak, relationType) &&
+            isAlreadyLinked(sourceZaak) &&
+            isAlreadyLinked(zrcClientService.readZaak(UUID.fromString(targetZaak.getObjectId())))
+
+    private fun isAlreadyLinked(zaak: Zaak) =
```

Comment:
Better to make a Kotlin extension function for this. Easier to read and use generally. See the examples in the previous PRs for this story.